examples/Custom_Hooks/probability_of_improvement.py

```diff
@@ -0,0 +1,195 @@
+## Example for calculating and plotting the Probability of Improvement
+
+# This example demonstrates how the {func}`register_hooks <baybe.utils.basic.register_hooks>` utility could be used to extract the probability of improvement (PI) from a running campaign:
+# * We define a hook that is compatible with the {meth}`BotorchRecommender.recommend <baybe.recommenders.pure.bayesian.BotorchRecommender.recommend>` interface and let's us extract the PI achieved after each experimental iteration,
+# * attach the hook to the recommender driving our campaign,
+# * and plot the evolving PI values after campaign completion.
+
+
+### Imports
+
+import os
+from types import MethodType
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+from botorch.test_functions.synthetic import Hartmann
+from matplotlib.collections import PolyCollection
+from mpl_toolkits.mplot3d import Axes3D
+from scipy.stats import gaussian_kde
+
+from baybe.acquisition import ProbabilityOfImprovement
+from baybe.campaign import Campaign
+from baybe.objectives.base import Objective
+from baybe.objectives.single import SingleTargetObjective
+from baybe.parameters import NumericalDiscreteParameter
+from baybe.recommenders import (
+    BotorchRecommender,
+    RandomRecommender,
+    TwoPhaseMetaRecommender,
+)
+from baybe.searchspace import SearchSpace, SearchSpaceType
+from baybe.surrogates import GaussianProcessSurrogate
+from baybe.targets import NumericalTarget
+from baybe.utils.basic import register_hooks
+from baybe.utils.botorch_wrapper import botorch_function_wrapper
+from baybe.utils.dataframe import to_tensor
+from baybe.utils.plotting import create_example_plots
+from baybe.utils.random import set_random_seed
+
+### Parameters for a full simulation loop
+
+# For the full simulation, we need to define some parameters.
+
+SMOKE_TEST = "SMOKE_TEST" in os.environ
+N_DOE_ITERATIONS = 3 if SMOKE_TEST else 7
+BATCH_SIZE = 2
+DIMENSION = 3
+POINTS_PER_DIM = 2 if SMOKE_TEST else 4
+
+# We also define the random seed to avoid having different plots.
+
+set_random_seed(1282)
+
+### Setup
+
+# We initialize a container for storing the PI from each recommend iteration:
+
+pi_per_iteration: list[np.ndarray] = []
+
+# Then, we define the hook that calculates the PI from each iteration.
+# To attach the hook we need to match its signature to that of {meth}RecommenderProtocol.recommend <baybe.recommenders.base.RecommenderProtocol.recommend>.
```

Comment:
please check if this link is correctly written

Reply:
The `` were missing. Thank you!